README.md

```diff
@@ -636,6 +636,29 @@ and caveats noted above apply.
 up to and including 3.1.0 or previous. The 2-tuple form wasn't added until
 3.2.0.
 
+Setting dynamic metadata at compile-time
+----------------------------------------
+Lager supports supplying metadata from external sources. You can add these by 
+using the `{lager_parse_transform_functions, X}` option. In rebar, you can 
+add it to `erl_opts`:
+
+```erlang
+{erl_opts, [{parse_transform, lager_transform}, 
+            {lager_function_transforms, 
+              [{metadata_name, {module_name, function_name}]}.
+```
+
+The MF called should take no arguments and should return a value that can be be
+ formatted into a log message. Any function which is not defined, loaded, or
+ throws an exception will return `undefined` or the provided default value.
+
+This metadata is also persistent across processes.
+ 
+**IMPORTANT**: Since you can calling functions, be aware that using these 
```

Comment:
Suggest an edit like

Since this feature relies on function calls injected at the point where a log may or may not be generated, your logging performance (ops/sec) will be impacted by what the functions you call do and how much latency they may introduce.